### DISCUSSION TITLE
Editieren: Automatische Attributierungen über räumliche Beziehungen

### DISCUSSION THREAD

**Discussion by k-placer** [e-netze/webgis-community#21] [General]
Hallo Jürgen,

Wenn ich die Option "Autovalue" beim Editieren  zur automatischen Attributierung über ber räumliche Beziehungen zu anderen Featureklassen konfiguriere, erhalte ich folgende Fehlermeldung:
`Service not found `

Ich habe unter https://docs.webgiscloud.com/cloud/tutorials/cms/editing/fields_autovalues.html nachgelesen:
          `TYP FROM kasten service strom@mycms BUFFERDIST 20 SEPERATOR space-space MAX 1`

`TYP`
Spalten-Name in der Datenbank? Oder Suchfeld Url? Oder Suchfeld Name?

`Kasten`
Name der Abfrage? Wäre das bei mir dann Name/Bezeichnung „Grundstück“ oder Url „grundstueck“

`service`
muss man tatsächlich das Wort „service“ eintippen?

`strom@mycms`
- 'strom' ist die Url des Diensts im CMS? 
- 'mycms' ist der Namensteil in der api.config d.h. <add key="cmspath_**mycms**"...> ?

LG K.

**Comment by gis-eni** [comment]
Die Fehlermeldung finde ich seltsam. Wenn ich genau dieses Query-String von oben bei mir in ein CMS kopiere bekomme ich den Fehler ``Service strom@mycms not found``. Das ist auch klar, weil es den Dienst bei mir nicht gibt. Seltsam ist, das der Servicename bei mir dezidiert in der Fehlermeldung steht und bei dir anscheind nicht...

Ist das wirklich die genau Fehlermeldung: ``Service not found``?

Wenn ich dann bei mir einen Dienst eingebe, den es auch wirklich gibt, funktioniert es.

Frage:
1. welchen Build verwendest du?
2. Ist der Dienst ``strom@mycms`` in irgend einer Art im CMS für den Zugriff beschränkt? Nicht für jeden Nutzer sichtbar?
3. Wenn ja, hat dein aktueller Benutzer Zugriff zu diesem Dienst?

**Reply by k-placer** [reply]
Mir war nicht klar war, welcher Parameter im Autovalue-Ausdruck, welchem Eintrag im CMS entspricht.
So funktioniert es bei mir:

`TYP FROM kasten service strom@mycms`
<table>
<tr><td>TYP</td><td>Spalten Name der Datenbank Tabelle </td></tr>
<tr><td>kasten</td><td>Name des Layers in ArcGIS Server Dienst</td></tr>
<tr><td>service </td><td>Konstante, muss immer in Autovalue-Ausdruck enthalten sein</td></tr>
<tr><td>strom@mycms</td><td>Dienst URL (nicht Dienst Name) in CMS mycms</td></tr>
</table>

**Reply by gis-eni** [reply]
Ich seh da jetzt keinen Unterschied zu dem was du vorher gemacht hast?! Aber wenns jetzt geht ;)

Ich glaub die Verwirrung kommt daher, dass ich das ``service`` in der Doku klein geschreiben haben. ``SERVICE`` ist eigentlich genau so ein Keyword wie ``BUFFERDIST`` oder ``MAX``. Hinter dem Keyword steht dann immer der entsprechende Wert. Wobei ``SERVICE`` (und natürlich ``FROM``) das einzige Keyword ist, dass seit WebGIS5 zwingend erforderlich ist.

``TYPE FROM kasten SERVICE strom@mycms BUFFERDIST 20...`` ist eventuell verständlicher.
Werde ich in der Doku ausbessern.

**Reply by saltdorfer** [reply]
Danke für das schöne Erklären der Parameter :)

Funktioniert bei euch die BUFFERDIST auch auf Punktobjekten? 
Ich schaff eine räumliche Beziehung nur mit einen Snapping auf den Punktlayer (in meinem Fall Adressen), bei Linien- oder Flächenobjekten als Basis funktioniert es.
lg
Sabine

**Reply by gis-eni** [reply]
Wenn es mit Snapping funktioniert, hab ich den Fehler gefunden. Das Problem ist dann mit Build 5.23.501 behoben. (ab morgen verfügbar)
